### PR TITLE
Switch from reqwest to fetch

### PR DESCRIPTION
**Pull request by weikinhuang**

### CODE REVIEW

File: karma-common-config.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ module.exports = {
   frameworks: ['mocha', 'requirejs', 'sinon-chai'],
 
   files: [
+    'bower_components/es6-promise/promise',
```
Comment (hzoo) [inline karma-common-config.js]: can you use the node_modules one here?

Comment (weikinhuang) [inline karma-common-config.js]: maybe another time

File: test/specs/resource.js [hunk 2]
```diff
@@ -277,27 +281,27 @@ define(['chai-as-promised', 'lib/resource'], function(chaiAsPromised, Resource)
...
             var resource = new Resource(descriptor);
 
             server.respondWith('POST', '/v1/foo', function(req) {
-              expect(req.requestBody).to.equal('bar=10');
+              expect(req.requestBody).to.equal('{"bar":10}');
               req.respond.apply(req, responses.foo);
             });
 
-            return expect(resource.post({ bar: 10 })).to.become(result);
+            return expect(resource.post(JSON.stringify({ bar: 10 })).then(parseResponse)).to.become(result);
```
Comment (kaicataldo) [inline test/specs/resource.js]: Can turn `'{"bar":10}'` into a variable and use it in both places

Comment (weikinhuang) [inline test/specs/resource.js]: I could...

File: test/specs/resource.js [hunk 3]
```diff
@@ -306,11 +310,11 @@ define(['chai-as-promised', 'lib/resource'], function(chaiAsPromised, Resource)
             var resource = new Resource(descriptor);
 
             server.respondWith('POST', '/v1/foo?baz=10', function(req) {
-              expect(req.requestBody).to.equal('bar=10');
+              expect(req.requestBody).to.equal('{"bar":10}');
```
Comment (kaicataldo) [inline test/specs/resource.js]: Same as https://github.com/behance/hippo/pull/22/files#r72152760

File: test/specs/resource.js (hunk 2 above)
Comment (kaicataldo) [inline test/specs/resource.js]: Seems a little silly to do it two different ways, but not a huge deal to me.

**Comment by kaicataldo** [comment]
:+1:

**Comment by Attamusc** [comment]
👍

**Pull request merged**